lib/streamlit/delta_generator.py

```diff
@@ -797,17 +775,19 @@ def get_last_dg_added_to_context_stack() -> DeltaGenerator | None:
 
 def _prep_data_for_add_rows(
     data: Data,
-    delta_type: str,
-    add_rows_metadata: AddRowsMetadata,
-) -> tuple[Data, AddRowsMetadata]:
+    add_rows_metadata: AddRowsMetadata | None,
+) -> tuple[Data, AddRowsMetadata | None]:
     out_data: Data
 
-    # For some delta types we have to reshape the data structure
+    # For built-in chart commands we have to reshape the data structure
```

Comment:
nit: if this function is only relevant for chart commands but not generally for everything that uses arrow / add-row functionality as the comment says, I suggest to rename the function to sth. like `_prep_chart_data_for_add_rows` and perhaps move it to the new `built_in_chart_utils.py`

Reply:
We cannot easily fully move this method since its used in all add row cases, but I extracted the chart specific logic to `built_in_charts_utils`.

Reply:
alrighty, but is this function only relevant for chart types or in general?

Reply:
The migrated part is only relevant for built-in charts, but `_prep_data_for_add_rows` is called in all add row cases.